napari/layers/points/points.py

```diff
@@ -1288,6 +1321,30 @@ def _view_data(self) -> np.ndarray:
 
         return data
 
+    @property
+    def _view_text(self) -> np.ndarray:
+
+        if len(self._indices_view) > 0:
+            if self._text._mode in [TextMode.FORMATTED, TextMode.PROPERTY]:
+                text = self._text.text[self._indices_view]
+            else:
+                text = np.array([''])
+        else:
+            # if no points in this slice send dummy data
+            text = np.array([''])
+
+        return text
+
+    @property
+    def _view_text_coords(self) -> np.ndarray:
+        if self._text._mode in [TextMode.FORMATTED, TextMode.PROPERTY]:
+            text_coords = (
+                get_text_anchors(self._view_data) + self._text.translation
+            )
+        else:
+            text_coords = np.zeros((0, self.dims.ndisplay))
+        return text_coords
```

Comment:
Why don't these attributes live inside the text manager? Could they instead be methods on the text manager, like `slice_text()` with the necessary args?

Reply:
We can move all of this into a `TextManager` member function. We will still have to pass in `layer._view_data`, but I think that is inline with the current implementation.

Reply:
Done!